### PR TITLE
[Merged by Bors] - bevy::scene::Entity renamed to bevy::scene::DynamicEntity.

### LINKED ISSUE #3233: Document and rename bevy::scene::Entity
## How can Bevy's documentation be improved?

This type ([docs](https://docs.rs/bevy/latest/bevy/scene/struct.Entity.html),  [source](https://github.com/bevyengine/bevy/blob/main/crates/bevy_scene/src/dynamic_scene.rs#L17)) appears to be used to store reflected entities, along with their components in scenes. However, it is a) completely lacking documentation b) very confusingly named: `Entity` is one of our most important types, and this is closely-related but distinct in important ways.

At a glance, this seems to be a reflection-powered serializable representation of an entity and its components. The `entity: u32` field  is likely for use with `Entity::new()` (the standard `Entity` type), and the entire `DynamicScene` type is used to spawn groups of entities which contain a number of entities, each with their own run-time defined set of components. These `DynamicScenes` can be written to file, but not read from.

Later in the same file, [`EntityMap`](https://github.com/bevyengine/bevy/blob/cf221f9659127427c99d621b76c8085c4860e2ef/crates/bevy_ecs/src/entity/map_entities.rs) is used, which contains a map between entities, in the standard `bevy_ecs` sense. This naming convention seems to be fine.

My first impulse would be to call this `DynamicEntity`, but please, bikeshed me if you disagree.

### PR DESCRIPTION
**Pull request by r4gus**
Basic documentation added to bevy::scene::DynamicEntity and bevy::scene::DynamicScene.

# Objective

- Rename bevy::scene::Entity to bevy::scene::DynamicEntity
- fixes #3233

## Solution

- Renamed the struct as requested.
- Added basic documentation.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Wow, those are excellent docs for a new contributor! Thanks for tackling this.

I think I would cut the `# Arguments` section from these docs; we don't use that style elsewhere and by-and-large the arguments here are self-evident.

**Comment by r4gus** [comment]
Alright, I'll remove them.

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_scene/src/dynamic_scene.rs [hunk 1]
```diff
@@ -58,17 +70,27 @@ impl DynamicScene {
         scene
     }
 
+    /// Write the dynamic entities and their corresponding components to the given world.
+    ///
+    /// This method will return an `SceneSpawnError` if either the type of a
+    /// component or the component itself hasn't been registered.
```
Comment (Davier) [inline crates/bevy_scene/src/dynamic_scene.rs:76]: To be more precise, the two errors cases are when a type is not registered and when a type doesn't reflect the Component trait.

**Review by Davier — APPROVED** [review]

**Pull request closed without merging**